hooks/pre-commit

```diff
@@ -182,9 +182,24 @@ else
 fi
 echo "${reset}"
 
+echo "Running unit tests in changed directories... "
+for dir in $(git diff --cached --name-only --diff-filter ACM |grep -e ^pkg/ -e ^cmd/ |xargs -r dirname |sort |uniq); do
```

Comment:
I can be biased by the problem I had with shippable not fetch "my" builds recently. Put in a standalone script, or an option like "--staging" for hack/test-go.sh ?